xbmc/windowing/osx/WinSystemOSX.mm

```diff
@@ -627,83 +612,34 @@ static void DisplayReconfigured(CGDirectDisplayID display,
 
-    // create new view if we don't have one
-    if (!m_glView)
-      m_glView = [[OSXGLView alloc] initWithFrame:rect];
+    m_appWindow = m_appWindowController.window;
+    m_glView = m_appWindow.contentView;
+  });
 
-    OSXGLView* view = (OSXGLView*)m_glView;
+  [m_glView.getGLContext makeCurrentContext];
+  [m_glView.getGLContext update];
 
-    dispatch_sync(dispatch_get_main_queue(), ^{
-      appWindow = [[OSXGLWindow alloc] initWithContentRect:NSMakeRect(0, 0, m_nWidth, m_nHeight)
-                                                 styleMask:windowStyleMask];
-      NSString* title = [NSString stringWithUTF8String:m_name.c_str()];
-      appWindow.backgroundColor = NSColor.blackColor;
-      appWindow.title = title;
-      appWindow.titlebarAppearsTransparent = YES;
-      appWindow.titleVisibility = NSWindowTitleHidden;
-
-      NSWindowCollectionBehavior behavior = appWindow.collectionBehavior;
-      //! @todo actually implement fullscreen tilling and remove NSWindowCollectionBehaviorFullScreenDisallowsTiling
-      behavior |= NSWindowCollectionBehaviorFullScreenPrimary |
-                  NSWindowCollectionBehaviorFullScreenDisallowsTiling;
-      [appWindow setCollectionBehavior:behavior];
-
-      // associate with current window
-      [appWindow setContentView:view];
-
-      // set the window to the appropriate screen and screen position
-      if (screen)
-      {
-        if (m_bFullScreen)
-        {
-          [appWindow setFrameOrigin:screen.frame.origin];
-        }
-        else
-        {
-          // if there are stored window positions use that as the origin point
-          const int top = settings->GetInt(SETTING_WINDOW_TOP);
-          const int left = settings->GetInt(SETTING_WINDOW_LEFT);
-
-          NSPoint windowPos;
-          if (top != 0 || left != 0)
-          {
-            windowPos = NSMakePoint(left, top);
-          }
-          else
-          {
-            // otherwise center the window on the screen
-            windowPos =
-                NSMakePoint(screen.frame.origin.x + screen.frame.size.width / 2 - m_nWidth / 2,
-                            screen.frame.origin.y + screen.frame.size.height / 2 - m_nHeight / 2);
-          }
-          [appWindow setFrameOrigin:windowPos];
-        }
-      }
-    });
-
-    [view.getGLContext makeCurrentContext];
-    [view.getGLContext update];
+  NSScreen* currentScreen = [NSScreen mainScreen];
+  dispatch_sync(dispatch_get_main_queue(), ^{
+    // NSWindowController does not track the last used screen so set the frame coordinates
+    // to the center of the screen in that case
+    if (screen && currentScreen != screen)
+    {
+      [m_appWindow
+          setFrameOrigin:NSMakePoint(
+                             screen.frame.origin.x + screen.frame.size.width / 2 - m_nWidth / 2,
+                             screen.frame.origin.y + screen.frame.size.height / 2 - m_nHeight / 2)];
```

Comment:
```suggestion
                             NSMidX(screen.frame) - m_nWidth / 2,
                             NSMidY(screen.frame) - m_nHeight / 2)];
```

Reply:
Added on new commit since this is also used in another location (also rebased)
Thanks for the review